jaxrs/src/main/java/org/jboss/aerogear/unifiedpush/rest/registry/installations/InstallationRegistrationEndpoint.java

```diff
@@ -221,33 +230,52 @@ public Response registerInstallation(
     @Consumes(MediaType.APPLICATION_JSON)
     @Produces(MediaType.APPLICATION_JSON)
     @ReturnType("org.jboss.aerogear.unifiedpush.rest.EmptyJSON")
-    public Response increasePushMessageReadCounter(@PathParam("id") String pushMessageId,
-                           @Context HttpServletRequest request) throws IOException {
-
-        // find the matching variation:
-        final Variant variant = loadVariantWhenAuthorized(request);
-        if (variant == null) {
-            return create401Response(request);
-        }
-
-        if (pushMessageId != null) {
-            try (final InputStream props = this.getClass().getResourceAsStream("/kafka/producer.props")) {
-                final Properties properties = new Properties();
-                properties.load(props);
-
-                final Producer<String, String> producer = new KafkaProducer<>(properties);
-                producer.send(new ProducerRecord<String, String>("installationMetrics", pushMessageId, variant.getVariantID()));
-                producer.close();
-            }
-
-            // let's do update the analytics
-            metricsService.updateAnalytics(pushMessageId, variant.getVariantID());
-        } else {
-            return Response.status(Status.BAD_REQUEST).build();
-        }
-
-        return Response.ok(EmptyJSON.STRING).build();
-    }
+	public Response increasePushMessageReadCounter(@PathParam("id") String pushMessageId,
+			@Context HttpServletRequest request) throws IOException {
+
+		// find the matching variation:
+		final Variant variant = loadVariantWhenAuthorized(request);
+		if (variant == null) {
+			return create401Response(request);
+		}
+
+		if (pushMessageId != null) {
+
+			// start the producer and push a message to installation metrics
+			// topic
+			final Properties properties = ConfigurationUtils.loadProperties(KAFKA_PRODUCER_PROPERTIES_PATH);
+			final Producer<String, String> producer = new KafkaProducer<>(properties);
+			producer.send(new ProducerRecord<String, String>(KAFKA_INSTALLATION_TOPIC, pushMessageId,
+					variant.getVariantID()));
+			producer.close();
+
+			// start a consumer that reads from "istallationMetrics" and based
+			// on the information updates the analytics
+			// read only messages which weren't read from other consumers
+			final Properties consumerProperties = ConfigurationUtils.loadProperties(KAFKA_CONSUMER_PROPERTIES_PATH);
```

Comment:
Where do u think it is a good place to add it? I was thinking about "push-sender" module, we can add a new package "kafka", so that consumers and producers can be stored.